Tests/TestCreateFFmpegInteropMSSFromUri.cs

```diff
@@ -18,177 +18,163 @@
-            Assert.IsNull(FFmpegMSS);
+            // CreateFFmpegInteropMSSFromUri should throw if uri is blank with default parameter
+            try
+            {
+                CreateMSSFromUri(string.Empty, null);
+                Assert.IsTrue(false);
+            }
+            catch (Exception)
+            {
+                // Call threw as expected
+            }
+
+            // CreateFFmpegInteropMSSFromUri should throw if uri is blank with non default parameter
+            try
+            {
+                var config = new FFmpegInteropMSSConfig
+                {
+                    ForceAudioDecode = true,
+                    ForceVideoDecode = true
+                };
+
+                CreateMSSFromUri(string.Empty, config);
+                Assert.IsTrue(false);
+            }
+            catch (Exception)
+            {
+                // Call threw as expected
+            }
         }
 
         [TestMethod]
         public void CreateFromUri_Bad_Input()
         {
-            // CreateFFmpegInteropMSSFromUri should return null when given bad uri
-            FFmpegInteropMSS FFmpegMSS = FFmpegInteropMSS.CreateFFmpegInteropMSSFromUri("http://This.is.a.bad.uri", false, false);
-            Assert.IsNull(FFmpegMSS);
+            // CreateFFmpegInteropMSSFromUri should throw when given bad uri
+            try
+            {
+                CreateMSSFromUri("http://This.is.a.bad.uri", null);
+                Assert.IsTrue(false);
+            }
+            catch (Exception)
+            {
+                // Call threw as expected
+            }
         }
 
         [TestMethod]
         public void CreateFromUri_Default()
         {
-            // CreateFFmpegInteropMSSFromUri should return valid FFmpegInteropMSS object which generates valid MediaStreamSource object
-            FFmpegInteropMSS FFmpegMSS = FFmpegInteropMSS.CreateFFmpegInteropMSSFromUri(Constants.StreamingUriSource, false, false);
-            Assert.IsNotNull(FFmpegMSS);
-
-            // Validate the metadata
-            Assert.AreEqual(FFmpegMSS.AudioCodecName.ToLowerInvariant(), "aac");
-            Assert.AreEqual(FFmpegMSS.VideoCodecName.ToLowerInvariant(), "h264");
-
-            MediaStreamSource mss = FFmpegMSS.GetMediaStreamSource();
-            Assert.IsNotNull(mss);
+            // Create the MSS
+            MediaStreamSource mss = CreateMSSFromUri(Constants.StreamingUriSource, null);
 
             // Based on the provided media, check if the following properties are set correctly
-            Assert.AreEqual(true, mss.CanSeek);
-            Assert.AreNotEqual(0, mss.BufferTime.TotalMilliseconds);
+            Assert.IsTrue(mss.CanSeek);
+            Assert.IsTrue(mss.BufferTime.TotalMilliseconds > 0);
             Assert.AreEqual(Constants.StreamingUriLength, mss.Duration.TotalMilliseconds);
+
+            // TODO: Verify expected stream types (aac, h264)
```

Comment:
Multiple TODO for verifying expected stream types in these tests.

Reply:
I'll handle test updates in a future change.